### PR TITLE
Feature/openurl toc fixes

### PR DESCRIPTION
**Pull request by Steven-Eardley**
Added date parsing, and a query to check if a TOC exists before routing there. Basic documentation; and text changes for issue #652.

### CODE REVIEW

File: portality/models/openurl.py [hunk 1]
```diff
@@ -124,11 +119,32 @@ def get_result_url(self):
...
-                # for sure at this point whether the user will hit a valid ToC
-                # construct the toc url using the ident, plus volume and issue if they are available
-                jtoc_url = url_for("doaj.toc", identifier=ident, volume=self.volume, issue=self.issue if self.volume else None)
-
+                # If there request has a volume parameter, query for presence of an article with that volume
+                if self.volume:
+                    volume_query = deepcopy(TERMS_SEARCH)
+                    volume_query["size"] = 0
+
+                    issn_term = { "term" : { "index.issn.exact" : ident} }
```
Comment (richard-jones) [inline portality/models/openurl.py]: A couple of things could go wrong here that we should look at:

1. ident might not be an issn.  For some unknown reason we might have wound up with a journal id, in which case this match may fail when it should have succeeded.

2. Irrespective of (1), journal continuations may have up to 2 issns, and articles may list one or both of them, so a match for one issn won't cover all possible cases.  We need to ask the bibjson for the specific continuation we are interested in for all of its issns (bibjson.issns()), and use these here as part of a "terms" query (not a "term" query).

**Comment by richard-jones** [comment]
Nearly there, just one outstanding issue around the issns used to query the articles

**Comment by Steven-Eardley** [comment]
@richard-jones this is working fine now, although there's more that could be done - see line note.

**Comment by richard-jones** [comment]
Looks good, merged

**Pull request merged**